core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/entity/player/input/BedrockMovePlayer.java

```diff
@@ -175,11 +184,11 @@ static void translate(GeyserSession session, PlayerAuthInputPacket packet) {
             }
         } else if (positionChangedAndShouldUpdate) {
             if (isValidMove(session, entity.getPosition(), packet.getPosition())) {
-                if (!session.getWorldBorder().isPassingIntoBorderBoundaries(entity.getPosition(), true)) {
-                    CollisionResult result = session.getCollisionManager().adjustBedrockPosition(packet.getPosition(), isOnGround, packet.getInputData().contains(PlayerAuthInputData.HANDLE_TELEPORT));
-                    if (result != null) { // A null return value cancels the packet
-                        Vector3d position = result.correctedMovement();
+                CollisionResult result = session.getCollisionManager().adjustBedrockPosition(packet.getPosition(), isOnGround, packet.getInputData().contains(PlayerAuthInputData.HANDLE_TELEPORT));
+            if (result != null) { // A null return value cancels the packet
+                Vector3d position = result.correctedMovement();
 
+                if (!session.getWorldBorder().isPassingIntoBorderBoundaries(position.toFloat(), true)) {
```

Comment:
why has this check been moved? Seems easier to return early when a player is outside of the world